src/proto/streams/streams.rs

```diff
@@ -984,6 +984,8 @@ impl<B> StreamRef<B> {
         let mut me = self.opaque.inner.lock().unwrap();
         let me = &mut *me;
 
+        me.refs += 1;
```

Comment:
This will need to go down lower, perhaps right after `let opaque = OpaqueStreamRef::new(..)`. That's because there's some checks in the middle that might not allow the push promise, and so we shouldn't increase the `refs` then.